dask/dataframe/io/sql.py

```diff
@@ -133,6 +158,14 @@ def read_sql_table(table, uri, index_col, divisions=None, npartitions=None,
         cond = index <= upper if i == len(lowers) - 1 else index < upper
         q = sql.select(columns).where(sql.and_(index >= lower, cond)
                                       ).select_from(table)
-        parts.append(delayed(pd.read_sql)(q, uri, **kwargs))
+        parts.append(delayed(_rationalize_sql)(q, uri, head[:0], **kwargs))
```

Comment:
Perhaps use `meta` here instead of `head[:0]`, and set `meta=head[:0]` on line 124 instead.

Reply:
Alternatively you could use `dask.dataframe.utils.check_meta` to raise a nice assertion error if the metadata differs, and forgo coercion completely. I'm not sure if what approach is best here.